Server/mods/deathmatch/logic/CPlayerManager.cpp

```diff
@@ -10,43 +10,36 @@
  *****************************************************************************/
 
 #include "StdInc.h"
+#include <numeric>
+#include <random>
 #include "net/SimHeaders.h"
 
-CPlayerManager::CPlayerManager()
-{
-    // Init
-    m_pScriptDebugging = NULL;
-    m_ZombieCheckTimer.SetUseModuleTickCount(true);
-}
-
 CPlayerManager::~CPlayerManager()
 {
-    DeleteAll();
+    // Deallocate all players
+    for (CPlayer* pPlayer : m_Players)
+        delete pPlayer;
```

Comment:
This is dangerous if the player destructor manages to change the underlying container **m_Players** while you're iterating over it. The previous code doesn't have this issue:
```cpp
   while (!m_Players.empty())
        delete *m_Players.begin();
```

Reply:
Wow, nice catch. Thanks!

Reply:
Turns out its not that nice of a catch.
As per the docs of `google_sparse_set` `erase()` doesn't invalidate iterators. (Because the destructor of CPlayer does indeed call `RemoveFromList`, which modifies `m_Players`, but only by calling `erase()`)